src/integer_mod_q/mat_polynomial_ring_zq/arithmetic/sub.rs

```diff
@@ -0,0 +1,218 @@
+
+        let poly_ring_mat3 = poly_ring_mat1 - poly_ring_mat2;
+
+        let cmp_poly_mat = MatPolyOverZ::from_str("[[3  0 0 1, 0],[2  -3 -4, 2  1 1]]").unwrap();
+        let cmp_poly_ring_mat = MatPolynomialRingZq::from((&cmp_poly_mat, &modulus));
+        assert_eq!(cmp_poly_ring_mat, poly_ring_mat3);
+    }
+
+    /// Testing sub_safe.
+    #[test]
+    fn sub_safe() {
+        let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+        let poly_mat1 = MatPolyOverZ::from_str("[[3  0 1 1, 1  42],[0, 2  1 2]]").unwrap();
+        let poly_ring_mat1 = MatPolynomialRingZq::from((&poly_mat1, &modulus));
+        let poly_mat2 = MatPolyOverZ::from_str("[[2  0 1, 1  42],[2  3 4, 2  0 1]]").unwrap();
+        let poly_ring_mat2 = MatPolynomialRingZq::from((&poly_mat2, &modulus));
+
+        let poly_ring_mat3 = poly_ring_mat1.sub_safe(&poly_ring_mat2).unwrap();
+
+        let cmp_poly_mat = MatPolyOverZ::from_str("[[3  0 0 1, 0],[2  -3 -4, 2  1 1]]").unwrap();
+        let cmp_poly_ring_mat = MatPolynomialRingZq::from((&cmp_poly_mat, &modulus));
+        assert_eq!(cmp_poly_ring_mat, poly_ring_mat3);
+    }
+
+    /// Testing sub_safe throws an error if the dimensions mismatch.
+    #[test]
+    fn sub_safe_error_dim() {
+        let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+        let poly_mat1 = MatPolyOverZ::from_str("[[4  -1 0 1 1, 1  42],[0, 2  1 2]]").unwrap();
+        let poly_ring_mat1 = MatPolynomialRingZq::from((&poly_mat1, &modulus));
+        let poly_mat2 = MatPolyOverZ::from_str("[[3  3 0 1, 1  42, 0],[0, 1  17, 1  1]]").unwrap();
+        let poly_ring_mat2 = MatPolynomialRingZq::from((&poly_mat2, &modulus));
+        let poly_mat3 = MatPolyOverZ::from_str("[[3  3 0 1, 1  42, 0]]").unwrap();
+        let poly_ring_mat3 = MatPolynomialRingZq::from((&poly_mat3, &modulus));
+
+        assert!(poly_ring_mat1.sub_safe(&poly_ring_mat2).is_err());
+        assert!(poly_ring_mat3.sub_safe(&poly_ring_mat2).is_err());
+    }
+
+    /// Testing sub_safe throws an error if the moduli mismatch.
+    #[test]
+    fn sub_safe_error_modulus() {
+        let modulus1 = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+        let modulus2 = ModulusPolynomialRingZq::from_str("4  1 0 1 1 mod 17").unwrap();
+        let modulus3 = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 18").unwrap();
+        let poly_mat = MatPolyOverZ::from_str("[[4  -1 0 1 1, 1  42],[0, 2  1 2]]").unwrap();
+        let poly_ring_mat1 = MatPolynomialRingZq::from((&poly_mat, &modulus1));
+        let poly_ring_mat2 = MatPolynomialRingZq::from((&poly_mat, &modulus2));
+        let poly_ring_mat3 = MatPolynomialRingZq::from((&poly_mat, &modulus3));
+
+        assert!(poly_ring_mat1.sub_safe(&poly_ring_mat2).is_err());
+        assert!(poly_ring_mat3.sub_safe(&poly_ring_mat2).is_err());
+    }
+
+    /// Tests the doc test (availability).
+    #[test]
+    fn doc_test() {
```

Comment:
Consider renaming this test to availability. This way it is not a problem if this test and the doc test are not the same.